### DISCUSSION TITLE
ODBC Async support?

### DISCUSSION THREAD

**Discussion by arunnabraham** [reactphp/reactphp#477] [General]
I was looking for a way to use odbc connections for async io in ReactPHP, couldn't see any support on this. Not sure whether its easy to implement. This could be a great way to connect to multiple DB vendors who even dont have native PHP support yet.

**Comment by SimonFrings** [comment]
Hey @arunnabraham, there are multiple ways to achieve this but I think the easiest and quickest way is to take a look at https://github.com/clue/reactphp-pq. This turns blocking functions into non-blocking promises, so you're able to do stuff while you wait for your DB operations to finish. This is especially useful for ODBC as they can take up some additional time in contrast to directly interacting with the database.

Other ways would be looking into implementing this as part or ReactPHP or as a separate project, but this involves quite some work 😅

**Reply by arunnabraham** [reply]
Thanks @SimonFrings for your response. I usually don't think much about ODBC - until I encountered a case where many DB vendors support ODBC or even third-party support for that, but async for this solves many problems for people to migrate from DB to DB vendors with async in PHP. I know it's a big thing to do but something that needs much attention in the community.

BTW the one you mentioned about ReactPHP PQ is the one I was looking for while I was working to integrate an application using the ReactPHP ecosystem. I would like to see a better result on this. And many thanks for that :)
